autodiff-template

```diff
@@ -176,73 +177,80 @@ GradNum2 div(in GradNum2 a, in GradNum2 b)
 	GradNum2 r = GradNum2(a.val / b.val, a.g);
 	r.g = r.g - r.val*b.g;
 	r.g = r.g / b.val;
-	
 	return r;
-	
 }
+
 GradNum2 div(in GradNum2 a, float b)
 {
 	return GradNum2(a.val/b, a.g/b);
 }
 
+GradNum2 div(float a, in GradNum2 b)
+{
+	return div(a_constG2(a), b);
+}
+
 GradNum2 a_sin(in GradNum2 a)
 {
 	float c = cos(a.val);
 	float s = sin(a.val);
 	return GradNum2(s , c * a.g);
-
 }
 
 GradNum2 a_cos(in GradNum2 a)
 {
 	float c = cos(a.val);
 	float s = sin(a.val);
 	return GradNum2(c , -s * a.g);
-
 }
 
 GradNum2 a_exp(in GradNum2 a)
 {
 	float e = exp(a.val);
 	return GradNum2(e , e * a.g);
-
 }
 
 GradNum2 a_log(in GradNum2 a)
 {
 	float ai = 1./a.val;
 	return GradNum2(log(a.val) , ai * a.g);
-
 }
 
 GradNum2 a_pow(in GradNum2 a, float k)
 {
 	float ap = k*pow(a.val,k-1.);
 	return GradNum2(pow(a.val,k) , ap * a.g);
+}
+
+GradNum2 a_pow(in GradNum2 a, in GradNum2 k)
+{
+	return a_exp(mult(k, a_log(a)));
```

Comment:
This is the main thing that I wanted to run by you... This is kind of a hack to re-use existing autodiff functions to implement `a_pow` instead of doing the math. It's something I found in another autodiff library.

What do you think? Does this seem OK to you?